handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslClientContext.java

```diff
@@ -164,6 +166,15 @@ static OpenSslSessionContext newSessionContext(ReferenceCountedOpenSslContext th
                 throw new SSLException("unable to setup trustmanager", e);
             }
             OpenSslClientSessionContext context = new OpenSslClientSessionContext(thiz, keyMaterialProvider);
+            // Enable session caching by default
+            context.setSessionCacheEnabled(true);
+            if (sessionCacheSize > 0) {
+                context.setSessionCacheSize((int) Math.min(sessionCacheSize, Integer.MAX_VALUE));
+            }
+            if (sessionTimeout > 0) {
```

Comment:
Why only positive numbers? Zero is also a legit value.
Negative values should throw an `IllegalArgumetException`, consider verifying input parameters from public API of `SslContextBuilder` or `OpenSslClientContext` ctors.